core/src/main/java/org/mapstruct/TargetProperty.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * This annotation marks a <em>presence check method</em> parameter as a property name parameter.
```

Comment:
Make a link to `@Condition` and make the use of this annotation exclusive to methods annotated with `@Condition`

Reply:
Also explicitly limit the type to `String` in this text. Perhaps have a testcase in which you use another java type and see if you get a decent error. Add error handling?

Reply:
1. Done
2. Not sure if I'm capable to make correct error handling.

Reply:
We always try to be as specific to users with our error messages. However, I do agree, we can do this enhancement in an additional PR.

Reply:
Ok.. I can live with that as well.. See #2863